lib/src/requests.dart

```diff
@@ -83,13 +83,22 @@ class Requests {
     Map<String, String> cookies = {};
     for (var key in responseHeaders.keys) {
       if (Common.equalsIgnoreCase(key, 'set-cookie')) {
-        String cookie = responseHeaders[key];
-        cookie
-            .split(';')
-            .map((x) => Common.split(x.trim(), '=', max: 1))
-            .where((x) => x.length == 2)
-            .where((x) => !_cookiesKeysToIgnore.contains(x[0].toLowerCase()))
-            .forEach((x) => cookies[x[0]] = x[1]);
+        var cookie = responseHeaders[key].trim();
+
+        var separators = [',', ';', ' '];
+        var lastSeparator = -1, endSeparator = 0;
+        cookie.split('').asMap().forEach((i, char) {
+          if(separators.contains(char)) lastSeparator = i;
+
+          if(char == '=' && i > endSeparator) {
+            endSeparator = cookie.indexOf(';', i) > 0 ? cookie.indexOf(';', i) : cookie.length;
+            var value = cookie.substring(i+1, endSeparator);
+            var key = cookie.substring(lastSeparator+1, i).trim();
+
+            if(_cookiesKeysToIgnore.contains(key.toLowerCase())) endSeparator = 0;
+            else cookies[key] = value;
+          }
+        });
```

Comment:
Seems inefficient to go over each character. 
Why not using .split with a regex?

```dart

var separators = [',', ';', ' '];
cookie
    .split(new RegExp(separators.join('|')))
    .map((x) => Common.split(x.trim(), '=', max: 1))
    .where((x) => x.length == 2)
    .where((x) => !_cookiesKeysToIgnore.contains(x[0].toLowerCase()))
    .forEach((x) => cookies[x[0]] = x[1]);

```

Reply:
I don't know much about Regex but I don't think that what the code does could be replicated by one.

Reply:
You implemented a multi-character `.split` logic (which is indeed needed here).
See suggested code below, same logic but with `split(new RegExp(",|;| "))` - which will split on either `,`, ` `, `;`

```
var separators = [',', ';', ' '];
cookie
    .split(new RegExp(separators.join('|')))
    .map((x) => Common.split(x.trim(), '=', max: 1))
    .where((x) => x.length == 2)
    .where((x) => !_cookiesKeysToIgnore.contains(x[0].toLowerCase()))
    .forEach((x) => cookies[x[0]] = x[1]);
```

BTW - Regular Expressions are awesome. I remember not understanding them as well in the past. If you want a 1on1 session about regex - mail me to jossef12@gmail.com. It's very useful once you catch it

Reply:
It's not a simple multi-character split because it has to tell apart those separators that are part of the cookie and those that are not. It cycles through the single chars to find an equal sing and from there decides if the cookie is meant to be parsed or not.

Oh, thanks, it's really kind of you but at the moment I'm all caught up with university and don't have much time unfortunately.